### PR TITLE
reduce diff with upstream puppetserver configs

### PR DESCRIPTION
**Pull request by mmoll**
this also prepares support for native OS packages of puppetserver

### CODE REVIEW

File: spec/classes/puppet_server_puppetserver_spec.rb [hunk 1]
```diff
@@ -497,12 +500,12 @@
 
         it 'should put the correct ssl key path in webserver.conf' do
           should contain_file('/etc/custom/puppetserver/conf.d/webserver.conf').
-            with_content(/ssl-key\s+=\s\/etc\/custom\/puppet\/ssl\/private_keys\/puppetserver43\.example\.com\.pem/)
+            with_content(/ssl-key:\s\/etc\/custom\/puppet\/ssl\/private_keys\/puppetserver43\.example\.com\.pem/)
```
Comment (ekohl) [inline spec/classes/puppet_server_puppetserver_spec.rb]: Maybe test using ```%r{}``` so you don't have to escape the slashes?

File: spec/classes/puppet_server_puppetserver_spec.rb [hunk 2]
```diff
@@ -497,12 +500,12 @@
...
         end
 
         it 'should put the correct ssl cert path in webserver.conf' do
           should contain_file('/etc/custom/puppetserver/conf.d/webserver.conf').
-            with_content(/ssl-cert\s+=\s\/etc\/custom\/puppet\/ssl\/certs\/puppetserver43\.example\.com\.pem/)
+            with_content(/ssl-cert:\s\/etc\/custom\/puppet\/ssl\/certs\/puppetserver43\.example\.com\.pem/)
```
Comment (ekohl) [inline spec/classes/puppet_server_puppetserver_spec.rb]: Same

File: spec/classes/puppet_server_puppetserver_spec.rb (hunk 1 above)
Comment (mmoll) [inline spec/classes/puppet_server_puppetserver_spec.rb]: done.

**Comment by domcleal** [comment]
Thanks @mmoll!

**Pull request merged**